core/services/ardupilot_manager/mavlink_proxy/AbstractRouter.py

```diff
@@ -79,6 +81,12 @@ def start(self, vehicle_endpoint: Optional[Endpoint] = None, _verbose: bool = Fa
         # pylint: disable=consider-using-with
         self._subprocess = subprocess.Popen(shlex.split(command), shell=False, encoding="utf-8", errors="ignore")
 
+        # Since the process takes some time to successfully start or fail, we need to wait before checking it's state
+        time.sleep(1)
+        if not self.is_running():
+            exit_code = self._subprocess.returncode
```

Comment:
That's a good idea.

With current implementation, this would require somewhat cumbersome to do, since the best way (I think) would be to do this asynchronously, using [asyncio subprocess](https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.asyncio.subprocess.Process.wait) and also adding the restart logic there.

In one of the next commits, which will enter the following PR, I already make the manager-start async, making this much easier to accomplish, so I suggest we stay like that for now and introduce this better-logging on the next PR.

What do you think?

Reply:
ping @rafaellehmkuhl

Reply:
Forgot to commit it. Done.